django/contrib/postgres/fields/array.py

```diff
@@ -28,6 +28,8 @@ def __init__(self, base_field, size=None, **kwargs):
         if self.size:
             self.default_validators = self.default_validators[:]
             self.default_validators.append(ArrayMaxLengthValidator(self.size))
+        if hasattr(self.base_field, 'from_db_value'):
```

Comment:
It would be helpful to add a comment like: "For performance, only add a from_db_value() method if the base field implements it."